rpc/src/v1/impls/personal_signer.rs

```diff
@@ -71,13 +68,18 @@ impl<A: 'static, C: 'static, M: 'static> PersonalSigner for SignerClient<A, C, M
 						if let Some(gas_price) = modification.gas_price {
 							request.gas_price = Some(gas_price);
 						}
-						match accounts.locked_account_secret(&request.from, &pass) {
-							Ok(secret) => {
-								let res = sign_and_dispatch(&*client, &*miner, request, secret);
-								queue.request_confirmed(id, res.clone());
-								Some(res)
+
+						let sender = request.from;
+						if let Err(_) = accounts.unlock_account_temporarily(sender, pass) {
```

Comment:
Unlikely, temp unlock unlocks account only for 1 signing. 

But, because this code may appear suspicious, I will alter it to sign the transaction directly, without unlocking.